File: packages/sms-mes-planning-pss-ui/src/services/pssApi/planningHorizon.ts
import { ApiEndpoints, ApiFunction } from '@sms/plasma-api-client';
import { PlanningHorizonTypes } from '../../types';

export interface ApiFunctions {
  getAllPlanningHorizon: ApiFunction<void, void, PlanningHorizonTypes.PlanningHorizon[]>;
  getComboPlanningHorizon: ApiFunction<
    { id: number },
    void,
    Omit<
      PlanningHorizonTypes.PlanningHorizon,
      'description' | 'horizon' | 'isDefault' | 'productionUnitGroupId' | 'materialAttributeDefinitionId'
    >[]
  >;
  createPlanningHorizon: ApiFunction<void, Omit<PlanningHorizonTypes.PlanningHorizon, 'id'>, void>;
  updatePlanningHorizon: ApiFunction<void, PlanningHorizonTypes.PlanningHorizon, void>;
  deletePlanningHorizon: ApiFunction<{ id: string | number }, void, void>;
}

const root = '/planningHorizon';

export const endpoints: ApiEndpoints<ApiFunctions> = {
  getAllPlanningHorizon: {
    method: 'GET',
    path: `${root}`,
  },
  getComboPlanningHorizon: {
    method: 'GET',
    path: `${root}/{id}`,
  },
  createPlanningHorizon: {
    method: 'POST',
    path: `${root}`,
  },
  updatePlanningHorizon: {
    method: 'PUT',
    path: `${root}`,
  },
  deletePlanningHorizon: {
    method: 'DELETE',
    path: `${root}/{id}`,
  },
};

File: packages/sms-mes-planning-pss-api/Models/PlanningHorizon.cs
namespace sms_mes_planning_pss_api.Models
{
  public class PlanningHorizon
  {
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int Horizon { get; set; }
    public bool IsDefault { get; set; }
    public int ProductionUnitGroupId { get; set; }
    public int MaterialAttributeDefinitionId { get; set; }
  }

  public class PlanningHorizonForQuery
  {
    public string AttributeName { get; set; }
    public int Horizon { get; set; }
  }
}
